src/geometry/CTiglBSplineApproxInterp.cpp

```diff
@@ -348,29 +363,55 @@ CTiglApproxResult CTiglBSplineApproxInterp::solve(const std::vector<double>& par
-            dy(intpIndex) = p.Y();
-            dz(intpIndex) = p.Z();
-            interpParams(intpIndex) = params[*it_idx];
-            intpIndex++;
+    if (n_intpolated + contin_cons > 0) {
+        // Write d vector. These are the points that should be interpolated as well as the continuity constraints for closed curve
+        math_Vector dx(1, n_intpolated + contin_cons, 0.);
+        math_Vector dy(1, n_intpolated + contin_cons, 0.);
+        math_Vector dz(1, n_intpolated + contin_cons, 0.);
+        if(n_intpolated > 0) {
+            TColStd_Array1OfReal interpParams(1, n_intpolated);
+            Standard_Integer intpIndex = 1;
+            for (std::vector<size_t>::const_iterator it_idx = m_indexOfInterpolated.begin(); it_idx != m_indexOfInterpolated.end(); ++it_idx) {
+                Standard_Integer ipnt = static_cast<Standard_Integer>(*it_idx + 1);
+                const gp_Pnt& p = m_pnts.Value(ipnt);
+                dx(intpIndex) = p.X();
+                dy(intpIndex) = p.Y();
+                dz(intpIndex) = p.Z();
+                interpParams(intpIndex) = params[*it_idx];
+                intpIndex++;
+            }
+            math_Matrix C = CTiglBSplineAlgorithms::bsplineBasisMat(m_degree, flatKnots, interpParams);
+            math_Matrix Ct = C.Transposed();
+            lhs.Set(1, nCtrPnts, nCtrPnts + 1, nCtrPnts + n_intpolated, Ct);
+            lhs.Set(nCtrPnts + 1,  nCtrPnts + n_intpolated, 1, nCtrPnts, C);
         }
 
-        math_Matrix C = CTiglBSplineAlgorithms::bsplineBasisMat(m_degree, flatKnots, interpParams);
-        math_Matrix Ct = C.Transposed();
-
-        lhs.Set(1, nCtrPnts, nCtrPnts + 1, nCtrPnts + n_intpolated, Ct);
-        lhs.Set(nCtrPnts + 1,  nCtrPnts + n_intpolated, 1, nCtrPnts, C);
-
+        // sets the C2 continuity constraints for closed curves on the left hand side if requested
+        if(is_closed && C2_continuous) {
+            math_Matrix continuity_entries(1,contin_cons,1,nCtrPnts);
+            continuity_entries.Init(0.);
+            TColStd_Array1OfReal continuity_params1(1, 1);
+            TColStd_Array1OfReal continuity_params2(1, 1);
+            continuity_params1(1) = params[0];
+            continuity_params2(1) = params[params.size()-1];
+            math_Matrix diff1_1 = CTiglBSplineAlgorithms::bsplineBasisMat(m_degree, flatKnots, continuity_params1,1);
+            math_Matrix diff1_2 = CTiglBSplineAlgorithms::bsplineBasisMat(m_degree, flatKnots, continuity_params2,1);
+            math_Matrix diff1 = diff1_1 - diff1_2;
+            math_Matrix diff2_1 = CTiglBSplineAlgorithms::bsplineBasisMat(m_degree, flatKnots, continuity_params1,2);
+            math_Matrix diff2_2 = CTiglBSplineAlgorithms::bsplineBasisMat(m_degree, flatKnots, continuity_params2,2);
+            math_Matrix diff2 = diff2_1 - diff2_2;
+            continuity_entries.Set(1,1,1,nCtrPnts,diff1);
+            continuity_entries.Set(2,2,1,nCtrPnts,diff2);
+            if(first_and_last ==false) {
```

Comment:
fix style. Space after if, space after ==

You can use astyle to create a correct style (see tigl wiki: https://github.com/DLR-SC/tigl/wiki/TiGL-Programmers-Guide#automatic-code-formatting-to-tigl-style-guide)